sdk/nodejs/servicebus/zMixins.ts

```diff
@@ -305,8 +305,19 @@ export class ServiceBusFunction extends appservice.Function<ServiceBusContext, s
             throw new Error("[subscription] must be specified in combination with [topic]");
         }
 
-        const namespaceName = (args.queue && args.queue.namespaceName) || args.topic!.namespaceName;
-        const resourceGroupName = (args.queue && args.queue.resourceGroupName) || args.topic!.resourceGroupName;
+        let namespaceName;
+        if (args.queue && args.queue.namespaceName) {
+            namespaceName = pulumi.all([args.queue.namespaceName, args.queue.id]).apply(([ns]) => ns);
+        } else {
+            namespaceName = pulumi.all([args.topic!.namespaceName, args.topic!.id]).apply(([ns]) => ns);
+        }
+
+        let resourceGroupName;
+        if (args.queue && args.queue.resourceGroupName) {
+            resourceGroupName = pulumi.all([args.queue.resourceGroupName, args.queue.id]).apply(([rg]) => rg);
+        } else {
+            resourceGroupName = pulumi.all([args.topic!.resourceGroupName, args.topic!.id]).apply(([rg]) => rg);
+        }

```

Comment:
I think I'd rather keep the old definitions and add another `const id = ...` to point to the queue/topic id, then pass the `id` to the existing `pulumi.all` call down below. Feels like it could be a bit simpler.

Reply:
Done.